languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -23318,6 +23318,20 @@ USA
     <!-- Grammar -->
     <!-- ====================================================================== -->
     <category id="GRAMMAR" name="Grammar" type="grammar">
+        <rule id="CONFUSION_OF_WHEN_WHAT" name="confusion of when/what" default="temp_off">
+            <pattern>
+                <marker>
+                    <token>when</token>
+                </marker>
+                <token postag="MD"/>
+                <token>be</token>
+                <token postag="VBN"/>
+                <token postag="IN"/>
+            </pattern>
+            <!-- concerned this might also falsely flag relative pronouns -->
```

Comment:
If you're concerned about relative pronouns, a solution might be to restrict the rule so that no noun phrases appear before "when". This would also take care of the one FP I see in the editor.
You can try writing an antipattern like how I showed you, using chunks. The chunks for noun phrases are:

`B-NP-singular` (beginning of singular NP)
`I-NP-singular` (middle of singular NP)
`E-NP-singular` (end of singular NP)
`B-NP-plural` (beginning of plural NP)
`I-NP-plural` (middle of plural NP)
`E-NP-plural` (end of plural NP)